engine/kafka/src/main/scala/pl/touk/nussknacker/engine/kafka/validator/TopicsExistenceValidator.scala

```diff
@@ -22,84 +22,78 @@ trait TopicsExistenceValidator {
+  private def config = kafkaConfig.topicsExistenceValidationConfig.validatorConfig
   private lazy val autoCreateSettingCache = new SingleValueCache[Boolean](expireAfterAccess = None, expireAfterWrite = Some(config.autoCreateFlagFetchCacheTtl))
-  private lazy val topicListCache = new SingleValueCache[Set[String]](expireAfterAccess = None, expireAfterWrite = Some(config.topicsFetchCacheTtl))
-  private lazy val kafkaAdminClient = {
-    val properties = new Properties()
-    properties.setProperty("bootstrap.servers", kafkaConfig.kafkaAddress)
-    logger.info(kafkaConfig.kafkaAddress)
-    AdminClient.create(properties)
-  }
-
-  def validateTopics(topics: List[String]): Validated[TopicExistenceValidationException, List[String]] =
-    isAutoCreateEnabled
-      .andThen(if (_) Valid(topics) else {
-        catchedErrorToInvalid {
-          topicListCache.getOrCreate {
-            kafkaAdminClient.listTopics(new ListTopicsOptions().timeoutMs(config.adminClientTimeout.toMillis.toInt)).names().get().asScala.toSet
-          }
-        }.andThen { existingTopics =>
-          val notExistingTopics = topics.diff(existingTopics.toList)
-
-          if (notExistingTopics.isEmpty)
-            Valid(topics)
-          else
-            Invalid(TopicExistenceValidationException.notExistingTopics(notExistingTopics))
+  private lazy val topicListCache = new SingleValueCache[List[String]](expireAfterAccess = None, expireAfterWrite = Some(config.topicsFetchCacheTtl))
+
+  def validateTopics(topics: List[String]): Validated[TopicExistenceValidationException, List[String]] = {
+    if (!kafkaConfig.topicsExistenceValidationConfig.enabled || isAutoCreateEnabled) {
+      Valid(topics)
+    } else {
+      topicListCache.get().flatMap(existingTopics => {
+        if(topics.diff(existingTopics).isEmpty)
+          Some(Valid(topics))
+        else
+          None
+      }).getOrElse {
+        val existingTopics = usingAdminClient {
+          _.listTopics(new ListTopicsOptions().timeoutMs(config.adminClientTimeout.toMillis.toInt))
+            .names().get().asScala.toList
         }
-      })
-
+        topicListCache.put(existingTopics)
+        val notExistingTopics = topics.diff(existingTopics)
+        if (notExistingTopics.isEmpty)
+          Valid(topics)
+        else
+          Invalid(new TopicExistenceValidationException(notExistingTopics))
+      }
+    }
+  }
 
-  private def isAutoCreateEnabled: Validated[TopicExistenceValidationException, Boolean] = catchedErrorToInvalid {
-    autoCreateSettingCache.getOrCreate {
-      val timeout = config.adminClientTimeout.toMillis.toInt
-      val id = kafkaAdminClient
-        .describeCluster(new DescribeClusterOptions().timeoutMs(timeout)).nodes().get().asScala.head.id().toString
-      kafkaAdminClient
-        .describeConfigs(List(new ConfigResource(ConfigResource.Type.BROKER, id)).asJava,
+  private def isAutoCreateEnabled: Boolean = autoCreateSettingCache.getOrCreate {
```

Comment:
Can you add parameterless `()` so it will be visible that this is not clean function?